daq/test_modules/external_module.py

```diff
@@ -148,13 +148,16 @@ def _finalize(self):
             self.pipe = None
         return_code = self.host.terminate()
         LOGGER.info('%s test host finalize %s', self, return_code)
+        modified_code = return_code if self.test_name != 'fail' else not return_code
```

Comment:
What's the purpose of this? It's going to create so much confusion.

Reply:
This particular change is moving this check from somewhere else to here.  The purpose of the 'fail' module is to test that a failing module is handled correctly. Yes, it's awkward now (very old legacy test), but I'm not inclined to clean it up as part of this PR. I added a TODO to replace it with a proper unit test for a failing module.